File: src/sdk-reference/php/3/collection/search-specifications/snippets/search-specifications-1.js

<?php

use \Kuzzle\Kuzzle;

$filters = [
    "match_all" => [
        "boost" => 1
    ]
];

$options = [
  'from' => 0,
  'size' => 20
];

$kuzzle = new Kuzzle('localhost');
$dataCollection = $kuzzle->collection('collection', 'index');

try {
  $res = $dataCollection->searchSpecifications($filters, $options);
  
  foreach ($res['hits'] as $specification) {
    // Specification
  }
  
  // Total specifications count
  $res['total'];
}
catch (ErrorException $e) {

}

File: src/sdk/android/3/controllers/collection/search-specifications/snippets/search-specifications-1.java

import io.kuzzle.sdk.core.Kuzzle;
import io.kuzzle.sdk.core.Options; 

Kuzzle kuzzle = new Kuzzle("localhost");

JSONObject filters = new JSONObject()
   .put("match_all", new JSONObject()
       .put("boost", 1)
   );

Options options = new Options();
options.setFrom((long) 0);
options.setSize((long) 20);

kuzzle
  .collection("collection", "index")
  .searchSpecifications(filters, options, new ResponseListener<JSONObject>() {
    @Override
    public void onSuccess(JSONObject res) {
      for (int i = 0; i < res.getJSONArray("hits").length(); i++) {
        res.getJSONArray("hits").getJSONObject(i) // Specification
      }

      res.getString("total"); // Total specifications count
    }

    @Override
    public void onError(JSONObject error) {
      // Handle error
    }
  });
